### ISSUE TITLE
Incorporate AsyncAPI into Quarkus

### ISSUE THREAD

**Issue by kenfinnigan** [smallrye/smallrye#31]
Depends on https://github.com/smallrye/smallrye-async-api/issues/1

**Comment by ChMThiel** [comment]
@radcortez Any progress on this? ...would be great...

**Comment by radcortez** [comment]
Not at the moment. Are you using the SmallRye Async API currently in any way?

**Comment by ChMThiel** [comment]
We are using `quarkus-smallrye-reactive-messaging-kafka` and annotate our methods with `org.eclipse.microprofile.reactive.messaging.Incoming/Outgoing` to receive/send KafkaMessages like this:
```
    @Incoming("IN_TOPIC)
    @Outgoing("OUT_TOPIC)
    public OutData processSerialunitBookingEvent(InData inData) {
        //do some stuff
    }
```

It would be great if there is a way that quarkus auto-generates an asyncApi.yml for us, describing the topic/message/etc. (Like it is done with JaxRs-REST-APIs an @Schema-Anntoation etc.)

**Comment by radcortez** [comment]
Yes, that was the plan. Until now, we didn't get people asking for it. So let's see if we can move this forward.
